dpsim-models/include/dpsim-models/DP/DP_Ph3_SynchronGeneratorDQ.h

```diff
@@ -73,8 +73,8 @@ class SynchronGeneratorDQ : public MNASimPowerComp<Complex>,
 
   // #### MNA Functions ####
   /// Initializes variables of component
-  virtual void mnaCompInitialize(Real omega, Real timeStep,
-                                 Attribute<Matrix>::Ptr) override = 0;
+  void mnaCompInitialize(Real omega, Real timeStep,
+                         Attribute<Matrix>::Ptr leftVector) override = 0;
```

Comment:
why did you remove the virtual keyword?

Reply:
**TL;DR** I did due to a suggestion of sonar that said the `virtual` should be dropped because is redundant. Should I keep it?

Some reasoning follows, just to argue on why/what... According to the standard, apparently it is not needed [and from here](https://en.cppreference.com/w/cpp/language/virtual)

> If some member function vf is declared as virtual in a class Base, and some class Derived, which is derived, directly or indirectly, from Base, has a declaration for member function with the same
> 
> - name
> - parameter type list (but not the return type)
> - cv-qualifiers
> - ref-qualifiers
> Then this function in the class Derived is also virtual (whether or not the keyword virtual is used in its declaration) and overrides Base::vf (whether or not the word override is used in its declaration).


_Conditions:_ 
- this class `DP_Ph3_SynchronGeneratorDQ` inherits from `MNASimPowerComp`, and is parent to `DP_Ph3_SynchronGeneratorDQODE` and `DP_Ph3_SynchronGeneratorDQTrapez`
- `MNASimPowerComp.cpp` provides an empty implementation of the `virtual void mnaCompInitialize` 
https://github.com/sogno-platform/dpsim/blob/f07066ab9941dfcb35e4acaf7086168f2b3462fb/dpsim-models/src/MNASimPowerComp.cpp#L120-L124
- and is not marked as pure virtual in `MNASimPowerComp.h`  and only virtual (done by design to be optional, according to the help)
https://github.com/sogno-platform/dpsim/blob/f07066ab9941dfcb35e4acaf7086168f2b3462fb/dpsim-models/include/dpsim-models/MNASimPowerComp.h#L75-L76

_Intention of the code:_
-  if we keep the implementation optional (then the childs will still inherit from `MNASimPowerComp`: consecuence is then that lines 76-77 need to be deleted (because we are not overriding neither implementing here) but we would be able to get instances of `DP_Ph3_SynchronGeneratorDQ` 
- if we force the implementation in the subclass (the intention seems more logical to me is that we only want to allow subclasses): the virtual keyword is not needed (according to sonar) because is inherited from the functon in `MNASimPowerComp`, and then we avoid the instantiation of this current class and forcing the implementation of it in the childs by marking it as `pure virtual` with the override

Reply:
Update: I do not have a strong opinion to be in favour of sonar, I changed it.